client/apps/account-settings/view-wrapper.js

```diff
@@ -33,8 +33,12 @@ class LabelSettingsWrapper extends Component {
 	}
 
 	onSaveSuccess = () => {
-		const { noticeActions, translate, orderHref } = this.props;
-		const options = orderHref ? { button: translate( 'Back to order' ), href: orderHref } : { duration: 5000 };
+		const { noticeActions, translate, orderHref, paymentMethodSelected } = this.props;
+		const options =
+			orderHref && paymentMethodSelected
+				? { button: translate( 'Back to order' ), href: orderHref }
```

Comment:
Is it possible to make this contain the order number? E.g. "Back to Order #38"?

Reply:
Done — just parsed the post ID out of the URL. This could alternatively be done on the client to avoid passing around redundant data.